tools/certgen/internal/internal.go

```diff
@@ -27,11 +46,18 @@ const (
 
 // Params for creating an X.509 certificate and RSA private key pair for TLS.
 type Params struct {
-	PublicKeyPath    string
-	PrivateKeyPath   string
-	ValidityDuration time.Duration
-	Hostnames        string
-	RSAKeyLength     int
+	CertificateAuthority      bool
```

Comment:
I'm not a fan of how these Params struct is implemented and used.  Where data is transformed into the actual parameters to the process is inconsistent:
- Hostnames is stored as a string, but then accessed through GetHostnames, the only parameter to do so
- Certificate paths and data on the other hand are processed as part of the CreateCertificateAndPrivateKeyFiles.

If, eg, RootPublicCertificateData and RootPublicCertificatePath are both set, then one silently overrides the other.  If RootPublicCertificatePath and RootPrivateKeyData are set, it fails because the file "" can't be found.

This all obscures the flow of information and makes this harder to read, and makes the tool more prone to misuse.

These variables are all also duplicated for the flag parsing. (using the Var variants, eg flag.Int -> flag.IntVar would fix that.)

I think all of this code would be easier to understand if there was one section which explicitly took the flag values, and converted them into the proper types (checking for both __Path and __Data being set, erroring if both are set), and returning a type narrower than params.  Also converting hostnames.

Reply:
The flags are not the only inputs for Params. It'll be used in tests a lot going forward. I'll look into splitting out the file loading since I can see how that's confusing.

Reply:
Thanks for this feedback. I addressed most of the concerns here.

* File paths are removed from `Params`.
* `Hostnames` is now `[]string`.
* Fixed the headers.

I did not use *Var( flags. It's not clear what that's improving based on the feedback.

Reply:
I was mentioning the Var flags because the struct and flags had direct mapping, so creating the extra variables was unnecessary.  Now that they're not directly mapping, this is fine.